pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/validator/ModelValidatorImpl.groovy

```diff
@@ -386,10 +387,26 @@ class ModelValidatorImpl implements ModelValidator {
         boolean valid = true
 
         if (Jenkins.getInstance() != null) {
-            Descriptor desc = lookup.lookupFunctionFirstThenStep(meth.name)
             DescribableModel<? extends Describable> model
-            if (desc != null) {
-                model = lookup.modelForFunctionFirstThenStep(meth.name)
+
+            List<Class<? extends Describable>> parentDescribables = Utils.parentsForMethodCall(meth)
+
+            if (!parentDescribables.isEmpty()) {
+                model = parentDescribables.collect { p ->
```

Comment:
using `collectMany` should let you remove the `.find` at the end

Reply:
*facepalm*

Reply:
Actually, we still need that `find` to make sure we get the first non-null one we encounter. Eh, this code's ok. =)

Reply:
Then [findResult](http://docs.groovy-lang.org/2.4.3/html/groovy-jdk/java/util/Collection.html#findResult(groovy.lang.Closure)) is probably what you want :)